main.c

```diff
@@ -104,4 +135,118 @@ void gpio_config() {
+}
+
+/**
+ * @brief	Envía un byte a la salida stdout en forma bloqueante pero por poco tiempo
+ * @param	data    Byte o dato a transmitir
+ * @return	Nada
+ * @note    Define la salida estandar para la librería stdio
+ */
+void putch(char data) {
+    while (PIR1bits.TXIF == 0)   //Espera que haya espacio en la FIFO
+        continue;
+    TXREG = data;   //Envía el byte
+}
+
+/**
+ * @brief	Toma un byte de la entrada stdin en forma bloqueante
+ * @return	El byte recibido
+ * @note    Define la entrada estandar para la librería stdio
+ */
+char getch(void) {
+    while (PIR1bits.RCIF == 0)   //Espera hasta que haya un byte recibido
+        continue;
+    return RCREG;   //retorna lo recibido
+}
+
+/**
+ * @brief	Envía un byte a la salida stdout en forma bloqueante pero por poco tiempo
+ * @param	data    Byte o dato a transmitir
+ * @return	Nada
+ * @note    Define la salida estandar para la librería stdio
+ */
+void uart_tx_byte( uint8_t dato ) {
+    while (PIR1bits.TXIF == 0)   //Espera que haya espacio en la FIFO
+        continue;
+    TXREG = dato;   //Envía el byte
+}
+
+/**
+ * @brief       Toma un byte de la FIFO de recepción en forma no bloqueante,
+ *              chequea primero si hay un dato disponible
+ * @param[out]  dato    Apunta al destino para el byte recibido
+ * @return      1 si hay un byte recibido, 0 si no hay dato disponible 
+ */
+uint8_t uart_rx_byte( uint8_t *dato ) {
+    if (PIR1bits.RCIF == 1) {
+        *dato = RCREG;
+        return 1;
+    } else {
+        return 0;
+    }
+}
+void refreshScreen(){
+    int idx;
+    
+    max7219_clear_display(MAX_DISPLAY_0);
+    max7219_set_led(MAX_DISPLAY_0, ball.y, ball.x, LED_ON);
+        
+    for( idx = 0 ; idx < line.size ; idx++ ) {
+            max7219_set_led(MAX_DISPLAY_0, line.y + idx, line.x, LED_ON);
+        }
+        
+ }
+void calculateDir(){
+    if(ball.x == line.x + 1 && ball.dir_x < 0){ 
+            // Si la pelota va en direccion a la paleta y esta en la col anterior 
+        if(ball.y >= line.y && ball.y < line.y + line.size){    
+                // Rebota si coincide con algun punto de la paleta    
+                ball.dir_x *= -1;
+            }
+        if(ball.y == line.y - 1 && ball.dir_y > 0){
+                // Rebota si pega contra las esquinas de la paleta
+                ball.dir_x *= -1;
+                ball.dir_y *= -1;            
+            }
+         if(ball.y == line.y + line.size && ball.dir_y < 0){
+                // Rebota si pega contra las esquinas de la paleta
+                ball.dir_x *= -1;
+                ball.dir_y *= -1;        
+            }
+    }
+    
+    if(ball.y == 7 || ball.y == 0){
+            // Rebota cuando llega al limite superior e inferior
+            ball.dir_y *= -1;
+        }
+        if(ball.x == 7 || ball.x == 0){
+            // Rebota cuando llega al limite izquierdo o derecho
+            ball.dir_x *= -1;
+        }
+}
+void setup(){
+    ball.x = 7; 
```

Comment:
Lo que causa el problema es que la pelota arranque justo en el borde de la pantalla.
Si cambias esta línea te va a funcionar bien.

```suggestion
    ball.x = 6; 
```